### DISCUSSION TITLE
Enforce bounding box recomputation

### DISCUSSION THREAD

**Discussion by 0penBrain** [coin3d/coin#504] [General]
Hi there,

Writing here because I'm currently facing an issue that I can't work around.
In a project (namely [FreeCAD](https://github.com/FreeCAD/FreeCAD), we have a [custom class](https://github.com/FreeCAD/FreeCAD/commits/master/src/Gui/SoDatumLabel.h) that subclasses SoShape.

The problem there is that the bounding box of the shape is depending on its rendering (more precisely, the bounding box is varying depending on the camera position/orientation/focal/zoom, as well as the rendering is).

By tracing the code calls, I found that `computeBBox` isn't called for example on zooming the view, which is a problem in our use case as the bounding box is changing.

Would there be a (direct or indirect) way to ensure that bounding box of shape is requested (recomputed) on each view change ?

I'm thinking to things like:
* Inform coin3D that our bounding box is "volatile" so it would call `computeBBox` on any camera change ?
* Forcefully push our new bounding box through an API (that we can call from our code in a camera node sensor) ?
* ...

Any help welcome. :smile:

TIA

**Comment by 0penBrain** [comment]
@VolkerEnderlein Sorry to ping but i'm not sure GH discussions are actively followed. :smile: 
May you have an idea ? Thanks.

**Reply by VolkerEnderlein** [reply]
I do not know what SoDatumLabel is used for in FreeCAD but I suppose it should display a kind of tool tip at a specific position and fixed in size on the screen.
There are two nodes in Coin that need to be scaled according to the view/screen properties: SoText2 and SoImage (which are also derived from SoShape).  I had a quick look onto SoText2 and it calculates a quad around the text and then projects the points of the quad onto the screen. Its bounding box is then extended to contain the quads points. Maybe that helps you to figure out what needs to be done for SoDatumLabel. Hope that helps further

**Reply by 0penBrain** [reply]
Thanks @VolkerEnderlein.
So far I have spent some time reading SoText2 and was unable to understand exactly what makes that its bounding box is recomputed on view (typically zoom) change and not our class.
There's something I'm still not able to understand there. Will continue looking at it.
